external/js/cothority/src/bevm/bevm-rpc.ts

```diff
@@ -0,0 +1,63 @@
+import Log from "../log";
+import { ServerIdentity, WebSocketConnection } from "../network";
+import {
+    CallRequest,
+    CallResponse,
+} from "./proto";
+
+/**
+ * BEvm service
+ */
+export class BEvmRPC {
+    static serviceName = "BEvm";
+
+    private conn: WebSocketConnection;
+    private timeout: number;
+
+    constructor(srvid: ServerIdentity) {
+        this.timeout = 60 * 1000; // 60 seconds
+        this.conn = new WebSocketConnection(srvid.getWebSocketAddress(),
+                                            BEvmRPC.serviceName);
+    }
+
+    /**
+     * Set a new timeout value for future requests.
+     *
+     * @param value Timeout in [ms]
+     */
+    setTimeout(value: number): void {
+        this.timeout = value;
+    }
+
+    /**
+     * Execute a view method (read-only)
+     *
+     * @param byzcoinId         ByzCoin ID
+     * @param bevmInstanceId    BEvm instance ID
+     * @param accountAddress    Address of the EVM account
+     * @param contractAddress   Address of the smart contract
+     * @param callData          ABI-packed call arguments
+     *
+     * @return Result of the view method execution
+     */
+    async performViewMethodCall(byzcoinId: Buffer,
```

Comment:
This would be clearer to me, but if you have a good reasoning for the current name, keep it... s/performViewMethodCall/callViewMethod/ ?

Reply:
In fact that's a good argument ;) I like when the go-services and the methods that call it have the same name. It's a manual grpc...

Reply:
Starting to see why grpc is better - it does this automatically what I ask you to do manually...

Here we have:
- bevm/api.go - CallEVM
- bevm/api.go - Client.PerformCall
- bevm/proto.go - CallRequest
- bevm/service.go - Service.PerformCall
- cothority/src/bevm/bevm-rpc.ts - BEvmRPC.callViewMethod

Compare this to something like calypso:
- calypso/api.go - Client.Authorize
- calypso/proto.go - Authorize
- calypso/service.go - Service.Authorize
- cothority/src/calypso/calypso-rpc.ts - OnChainSecretRPC.authorize

So if you have to trace the call across ts/go, I find the latter much easier to read... Unfortunately, if we change the `proto.go` definition, the popcoin app will stop working :( So if we decide to change the name here, we should inform them beforehand.

Reply:
Yes, that's one of the advantages of an RPC framework ;-)

I would say that `CallEVM()` is not related, as it is a function called by several places in BEvm, and just happens to be called by the service implementation of `PerformCall()`.

To me, the important link is indeed between the TS  service client and the Go service implementation function.
The name in `api.go` might not always be related, for example we could have a Client call that is implemented using two or more service calls (to various services even). Actually, I realize that `PerformCall()` in `api.go` is only used by the test, as a Go client will normally use `Call()` (symmetric to `BEvmInstance.call()`); maybe I should just move it to the test?

For the message name itself in `proto.go`, I agree, but there is a pair of messages (request and response). How about `PerformCallRequest` and `PerformCallResponse`? (btw, I see byzcoin code does not seem to follow the naming there, e.g. `AddTransaction()` <-> `AddTxRequest`)

By the way, the `proto.go` is already changed in the PR (removed methods, and `CallRequest`/`CallResponse` members changed). There will anyway be breaking changes for the Popcoin app as the TS API changes as well. But for now, the Popcoin app still refers to the "Stainless" conode for the service... so that should continue to work, right?

Reply:
1. `CallEVM` - true - that should not be on the list

2. `PerformCall` and `Call` - let's discuss this in the meeting this afternoon

3. Yes, and this is wrong :( Btw I stumble regularly to find the appropriate method for this call, now I know why...

4. Yes, at least for a grace-period. I'd ask them to move to the new API, and if you want you can do the same for your stainless-demo before you update the stainless conode

Reply:
Renamed as discussed during the meeting.